src/main/kotlin/org/jetbrains/research/ml/coding/assistant/util/psi/PsiUtil.kt

```diff
@@ -12,6 +12,7 @@ val PsiElement.isLeaf: Boolean
     get() = this.children.isEmpty()
 
 // TODO: we don't store "in" keyword for loops in labels. Is it ok?
+// TODO: it does not support async keyword now and type annotations
 val PsiElement.intermediateElementLabel: String
```

Comment:
Maybe we should add some description of why these types of PsiElement are considered and others are not? Is this a complete list of PsiElements we need to check?